dist/tools/openocd/openocd.sh

```diff
@@ -200,13 +202,14 @@ do_debug() {
             -c 'init' \
             -c 'targets' \
             -c 'halt' \
-            -l /dev/null" &
-    # save PID for terminating the server afterwards
-    OCD_PID=$?
+            -l /dev/null &> /dev/null & \
+            echo \$! > $tmp" &
     # connect to the GDB server
     ${DBG} ${TUI} -ex "tar ext :${GDB_PORT}" ${ELFFILE}
     # clean up
+    OCD_PID="$(cat $tmp)"
     kill ${OCD_PID}
+    rm $tmp
```

Comment:
use `trap 'rm -f $tmp' EXIT` near line 194 to always delete the PID file regardless of what happens later.